google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITJsonWriteReadTest.java

```diff
@@ -0,0 +1,156 @@
+  private static final String VALID_JSON_DIR = "valid";
+  private static final String INVALID_JSON_DIR = "invalid";
+
+  private static final String TABLE_NAME = "TestTable";
+
+  @ClassRule public static IntegrationTestEnv env = new IntegrationTestEnv();
+
+  private static DatabaseClient databaseClient;
+
+  @BeforeClass
+  public static void beforeClass() {
+    final RemoteSpannerHelper testHelper = env.getTestHelper();
+    final Database database =
+        testHelper.createTestDatabase(
+            "CREATE TABLE "
+                + TABLE_NAME
+                + "("
+                + "Id INT64 NOT NULL,"
+                + "json JSON"
+                + ") PRIMARY KEY (Id)");
+    databaseClient = testHelper.getDatabaseClient(database);
+  }
+
+  @Test
+  public void testWriteValidJsonValues() throws IOException {
+    assumeFalse("Emulator does not yet support JSON", EmulatorSpannerHelper.isUsingEmulator());
+
+    List<String> resources = getJsonFilePaths(RESOURCES_DIR + File.separator + VALID_JSON_DIR);
+
+    long id = 0L;
+    List<Mutation> mutations = new ArrayList<>();
+    List<String> jsons = new ArrayList<>();
+    for (String resource : resources) {
+      String jsonStr =
+          Resources.toString(
+              Resources.getResource(this.getClass(), VALID_JSON_DIR + File.separator + resource),
+              StandardCharsets.UTF_8);
+      jsons.add(jsonStr);
+      Mutation mutation =
+          Mutation.newInsertBuilder(TABLE_NAME)
+              .set("Id")
+              .to(id++)
+              .set("json")
+              .to(Value.json(jsonStr))
+              .build();
+      mutations.add(mutation);
+    }
+
+    databaseClient.write(mutations);
+    ResultSet resultSet =
+        databaseClient.singleUse().executeQuery(Statement.of("SELECT * FROM " + TABLE_NAME));
+    int count = 0;
+    while (resultSet.next()) {
+      count++;
+    }
+    assertEquals(count, resources.size());
+  }
+
+  @Test
+  public void testWriteAndReadInvalidJsonValues() throws IOException {
+    assumeFalse("Emulator does not yet support JSON", EmulatorSpannerHelper.isUsingEmulator());
+
+    List<String> resources = getJsonFilePaths(RESOURCES_DIR + File.separator + INVALID_JSON_DIR);
+
+    for (String resource : resources) {
+      String jsonStr =
+          Resources.toString(
+              Resources.getResource(this.getClass(), INVALID_JSON_DIR + File.separator + resource),
+              StandardCharsets.UTF_8);
+
+      try {
+        databaseClient.write(
+            Collections.singletonList(
+                Mutation.newInsertBuilder(TABLE_NAME)
+                    .set("Id")
+                    .to(100L)
+                    .set("json")
+                    .to(Value.json(jsonStr))
+                    .build()));
+        fail(resource + " should be rejected.");
+      } catch (Exception e) {
+        System.out.println("Failed to insert JSON: " + jsonStr + ". From: " + resource);
+        e.printStackTrace();
```

Comment:
Remove this and consider replacing it with an assertion that you actually got the error that you expected.